src/armature/Import.ts

```diff
@@ -0,0 +1,182 @@
+import { vec3 } from 'gl-matrix';
+
+import { RGBColor } from '../colors/RGBColor';
+import { Face, WorkingGeometry } from '../geometry/WorkingGeometry';
+import { Material } from '../renderer/Material';
+import { GeometryNode, Node } from './Node';
+
+/**
+ * A helper class that specifies a group as stored in the obj files
+ */
+class Group {
+    public groupName: string;
+    public materialName: string;
+    public faces: Face[];
+}
+
+/**
+ * A class to store the string and read information of a multiline string.
+ */
+class Data {
+    public lines: string[];
+
+    constructor(rawData: string) {
+        this.lines = rawData.split('\n');
+    }
+
+    public readLine() {
+        let line = this.lines[0].split(' ');
+        this.lines.shift();
+        // Skip comments and empty lines.
+        while (line.length < 1 || line[0].startsWith('#')) {
+            line = this.lines[0].split(' ');
+            this.lines.shift();
+        }
+
+        return line;
+    }
+
+    public getLinesWithPrefix(prefix: string) {
```

Comment:
I assume the thing that'll get this working for any arbitrary obj file is just taking out the assumption that they all come in a batch. A future refactor might be to pass in a function to handle a new vertex, face, etc and have this multiplex those?

Reply:
Added TODO.